lib/wifi_user/use_case/check_if_whitelisted_email.rb

```diff
@@ -0,0 +1,21 @@
+module WifiUser
+  module UseCases
+    class CheckIfWhitelistedEmail
+      def initialize(gateway:)
+        @gateway = gateway
```

Comment:
can this be a bit more descriptive as to what is expected, please?
maybe `email_regex_gateway`?